### ISSUE TITLE
breakfix: bypass URI not working

### ISSUE THREAD

**Issue by abbekeff** [greenpau/caddy-security#44]
> A clear and concise description of what you want to accomplish.

I want to bypass /api and some other paths from my policy 

Before with caddy-auth I had this in my caddyfile; (worked fine) 
```
qbit.website.com {
    import tls
    authorize {
        bypass uri prefix /api
        bypass uri prefix /command
        bypass uri prefix /query
        bypass uri prefix /login
        bypass uri prefix /sync
        set auth url https://auth.website.com
        allow roles authp/admin
		validate bearer header
		inject headers with claims
    }
    encode zstd gzip
    import headers	
    reverse_proxy qbittorrent:8080
}
```

Now with caddy-security Im trying to achieve the same thing, so I looked here: 
https://authp.github.io/docs/authorize/bypass

```
    security {
		authentication portal myportal {
	     crypto default token lifetime 86400
             cookie lifetime 172800
             cookie domain website.com
			 backend local /config/auth/local/users2.json local
             ui {
                 links {
                      "Flame" https://flame.website.com/ icon "las la-star"
                 }
             }
			 transform user {
                match email email@gmail.com
				action add role authp/admin	
			}
        }
		authorization policy admins_policy {
            bypass uri prefix /api
            bypass uri prefix /command
            bypass uri prefix /query
            bypass uri prefix /login
            bypass uri prefix /sync
            allow roles authp/admin
            set auth url https://auth.website.com/
			acl rule {
				comment default deny
				match any
				deny log warn
			}
		}
    }
```

This however does not work, the reverse proxy stops working after auth is done (shows a blank page), also log spams with: 

`{"level":"error","ts":1643842456.2998343,"logger":"http.handlers.authentication","msg":"auth provider returned error","provider":"authorizer","error":"user authorization failed: src_ip=IP, src_conn_ip=IP, reason: user data not found"}`

`{"level":"error","ts":1643842194.8351867,"logger":"http.handlers.authentication","msg":"auth provider returned error","provider":"authorizer","error":"user authorization failed: src_ip=IP, src_conn_ip=IP, reason: no token found"}`

Am I not understanding the bypass function? Or is it a better way to do this? Thanks

**Comment by greenpau** [comment]
@abbekeff , please re-post your config after you ran formatting over it.

```
bin/caddy fmt -overwrite /path/to/Caddyfile
```

What is your current configuration for `qbit.website.com`?

Do you have `order` configured? (relevant explanation is https://github.com/greenpau/caddy-security/issues/34#issuecomment-1024293316)

```
order authenticate before respond
order authorize before basicauth
```

**Comment by abbekeff** [comment]
```
{
	http_port 8282
	https_port 4443
	order authenticate before respond
	order authorize before basicauth
	email "email"
	dynamic_dns {
		provider cloudflare API
		domains {
			website.com
		}
		check_interval 30m
		ip_source simple_http https://icanhazip.com
		ip_source simple_http https://api64.ipify.org
	}
	security {
		authentication portal myportal {
			crypto default token lifetime 86400
			cookie lifetime 172800
			cookie domain website.com
			backend local /config/auth/local/users2.json local
			ui {
				links {
					"Flame" https://flame.website.com/ icon "las la-star"
				}
			}
			transform user {
				match email "email"
				action add role authp/admin
			}
		}
		authorization policy admins_policy {
			bypass uri prefix /api
			bypass uri prefix /command
			bypass uri prefix /query
			bypass uri prefix /login
			bypass uri prefix /sync
			allow roles authp/admin
			set auth url https://auth.website.com/
			acl rule {
				comment default deny
				match any
				deny log warn
			}
		}
	}
}
(tls) {
	tls {
		issuer zerossl API{
			dns cloudflare API
			resolvers 1.1.1.1
		}
	}
}
(headers) {
	header {
		Strict-Transport-Security "max-age=31536000; includeSubDomains; preload"
		X-Frame-Options "SAMEORIGIN"
		X-Content-Type-Options "nosniff"
		Referrer-Policy "strict-origin"
		Content-Security-Policy "upgrade-insecure-requests"
		X-Robots-Tag "noindex, nofollow, nosnippet, noarchive"
	}
}

auth.website.com {
	log {
		output file /config/logs/access.log
		level WARN
	}
	import tls
	import headers
	route {
		authenticate with myportal
	}
}

flame.website.com {
	import tls
	encode zstd gzip
	import headers
	route {
		authorize with admins_policy
		reverse_proxy flame:5005
	}
}

radarr.website.com {
	import tls
	encode zstd gzip
	import headers
	route {
		authorize with admins_policy
		reverse_proxy radarr:7878
	}
}

sonarr.website.com {
	import tls
	encode zstd gzip
	import headers
	route {
		authorize with admins_policy
		reverse_proxy sonarr:8989
	}
}

prowlarr.website.com {
	import tls
	encode zstd gzip
	import headers
	route {
		authorize with admins_policy
		reverse_proxy prowlarr:9696
	}
}

qbit.website.com {
	import tls
	encode zstd gzip
	import headers
	route {
		authorize with admins_policy
		reverse_proxy qbittorrent:8080
	}
}

jellyfin.website.com {
	import tls
	import headers
	reverse_proxy jellynightly:8096 {
		flush_interval -1
	}
}

```

**Comment by greenpau** [comment]
@abbekeff , please enable `debug` in global config and try debugging. Please email me the logs.

Also, for troubleshooting, please configure static keys:

```
authentication portal myportal {
  crypto key sign-verify 39a7ed93-6341-4232-a4fe-c44d242eca7d
}

authorization policy admins_policy {
  crypto key verify 39a7ed93-6341-4232-a4fe-c44d242eca7d
}
```

**Comment by greenpau** [comment]
@abbekeff , for troubleshooting, amend your ACL:

from:

```
			allow roles authp/admin
			acl rule {
				comment default deny
				match any
				deny log warn
			}
```

To:

```
      acl rule {
        match role authp/admin
        allow stop log debug
      }
      acl rule {
        comment default deny
        match any
        deny log warn
      }
```

**Comment by greenpau** [comment]
Also, try without `route`:

```
flame.website.com {
	import tls
	encode zstd gzip
	import headers
        authorize with admins_policy
	reverse_proxy flame:5005
}
```

**Comment by abbekeff** [comment]
@greenpau
I tried all your suggestions, same issue still.  
I have logs for you, I cant find your email. where do I send them? (I'm new to github)

**Comment by greenpau** [comment]
greenpau@outlook.com

**Comment by greenpau** [comment]
Release fix in [v1.0.13](https://github.com/greenpau/caddy-security/releases/tag/v1.0.13).
